### PR TITLE
fix: #46 corrupt docker images pushed to docker hub, #69 bug in remote skill consumer mode

### LINKED ISSUE #46: Dockerhub images do not start
# Description

All images under 

https://hub.docker.com/search?q=tractusx%2Fagentplane-hashicorp

and

https://hub.docker.com/search?q=tractusx%2Fagentplane-azure-vault

currently do not start.

# Error

Error: Invalid or corrupt jarfile edc-dataplane.jar

# Should be

Image should start normally.

### LINKED ISSUE #69: Agent-Plane (all): Execution of Remote Skill in runMode/distributionMode=CONSUMER returns Skill Text instead of result.
# Observation

Invoking a remote skill through the agent-plane's matchmaking agent endpoint where either
- runMode=CONSUMER (and distributionMode=CONSUMER|BOTH)
- runMode=BOTH (and distributionMode=CONSUMER)

will return the text of the remote skill instead of a skill execution result.

# Wanted Behaviour

The skill should be executed in the consumers agent plane and the result should be returned.

# Since

Release 1.9.8

### PR DESCRIPTION
**Pull request by drcgjung**
## WHAT

fixes a bug where the published images are corrupt (contain no java binaries due to lacking --build-arg). We remove the build args for making sure that such a thing cannot happen anymore

furthermore, we flexibilize some workflows such that they can be manually triggered with options and such that they can be debugged on development repos for future releases.

Implements a lacking code path/shortcut when interacting with the DelegationService.
Tests the lacking code path.

## WHY

Currently, there is no working docker image in the official dockerhub repo.
#69 

## FURTHER NOTES

Closes #46 
Closes #69

### CODE REVIEW

**Review by the-tatanka — APPROVED** [review]
lgtm

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -89,9 +89,18 @@ jobs:
           GITHUB_ACTOR: ${{ github.actor }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      # Run Maven Deploy (if either running on main or a version tag on a fork)
```
Comment (carslen) [inline .github/workflows/build.yml:92]: As forks are potentially ephemeral upstream repository won't care about forks. You shall not implement fork specific steps into your workflows and push to the upstream repository, even if it's a catenax-ng fork. If you want to run workflows in your fork, then you have to mange it otherwize.

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -84,11 +84,20 @@ jobs:
           - agentplane-azure-vault
           - agentplane-hashicorp
     steps:
+
+      - name: Check github repository and set docker repo
+        id: set-docker-repo
+        run: |
+          [ "${{ github.repository }}" == "eclipse-tractusx/knowledge-agents-edc" ] && echo "REPO=tractusx" >> $GITHUB_OUTPUT
+          [ "${{ github.repository }}" == "eclipse-tractusx/knowledge-agents-edc" ] && echo "REPO=ghcr.io/catenax-ng/tx-knowledge-agents-edc" >> $GITHUB_OUTPUT
```
Comment (carslen) [inline .github/workflows/trivy.yml:92]: Same here, workflow shall not care about running in a fork.

File: .github/workflows/build.yml (hunk 1 above)
Comment (carslen) [inline .github/workflows/build.yml:92]: As forks are potentially ephemeral a upstream repository won't care about forks. You shall not implement fork specific steps into your workflows and push them to the upstream repository, even if it's a catenax-ng fork. If you want to run workflows in your fork, then you have to mange it otherwize.

File: .github/workflows/trivy.yml (hunk 2 above)
Comment (carslen) [reply .github/workflows/trivy.yml:92]: Sorry, order of comment is different than in PR review. Please refer to to the other comment.

Comment (drcgjung) [reply .github/workflows/trivy.yml]: thanks for the heads up, I think I found a way that now really focusses on eclipse-tractusx and does not reference any specific fork. Thanks for revisiting the PR ...

**Comment by drcgjung** [comment]
> LGTM, but I've ran dash tool on PR source (catenax-ng/tx-knowledge-agent-edc main branch) and I've determined changes in `DEPENDENCIES` file, which are not part of this PR. Please update DEPENDENCIES file and add to this PR.
> 
> After DEPENDENCIES has been updated, I'm fine to merge the PR.

DEPENDENCIES updated, still approved from what I saw. Sorry for mixing that bugfix into the PR, we will try to work on branches rather than main in the future.

**Review by carslen — APPROVED** [review]
LGTM

**Pull request merged**